### PR TITLE
Update "preexec" from "https://github.com/rcaloras/bash-preexec@master"

### LINKED ISSUE #2070: Enable autojump and direnv (and maybe others) plugins will result in wrong PROMPT_COMMAND syntax
## Expected Behavior

Work without error.

## Current Behavior

```shell
bash: PROMPT_COMMAND: line 3: syntax error near unexpected token `;'
bash: PROMPT_COMMAND: line 3: `_direnv_hook; ; autojump_add_to_database'
```

## Possible Solution

Change in vendor/github.com/rcaloras/bash-preexec/bash-preexec.sh:__bp_install()
from 
`existing_prompt_command="${PROMPT_COMMAND//$__bp_install_string[;$'\n']}" # Edge case of appending to PROMPT_COMMAND`
to
`existing_prompt_command="${PROMPT_COMMAND//$__bp_install_string+([;$ '\n'])}" # Edge case of appending to PROMPT_COMMAND`

## Context

The value $PROMPT_COMMAND:
```
# before load autojump
__bp_trap_string="$(trap -p DEBUG)" trap - DEBUG __bp_install
# after load autojump
__bp_trap_string="$(trap -p DEBUG)" trap - DEBUG __bp_install ; autojump_add_to_database

# before load direnv
__bp_trap_string="$(trap -p DEBUG)" trap - DEBUG __bp_install ; autojump_add_to_database
# after load direnv
_direnv_hook;__bp_trap_string="$(trap -p DEBUG)" trap - DEBUG __bp_install ; autojump_add_to_database
```

In vendor/github.com/rcaloras/bash-preexec/bash-preexec.sh
```
    local existing_prompt_command
    # Remove setting our trap install string and sanitize the existing prompt command string
    existing_prompt_command="${PROMPT_COMMAND//$__bp_install_string[;$'\n']}" # Edge case of appending to PROMPT_COMMAND
    # _direnv_hook;__bp_trap_string="$(trap -p DEBUG)" trap - DEBUG __bp_install ; autojump_add_to_database
    existing_prompt_command="${existing_prompt_command//$__bp_install_string}"
    # _direnv_hook; ; autojump_add_to_database
```

## Steps to Reproduce

1. `bash-it enable plugin direnv autojump`
2. `bash-it restart`
3. bash will complain:
```shell
bash: PROMPT_COMMAND: line 3: syntax error near unexpected token `;'
bash: PROMPT_COMMAND: line 3: `_direnv_hook; ; autojump_add_to_database'
```
4. `echo $PROMPT_COMMAND`
```
__bp_precmd_invoke_cmd _direnv_hook; ; autojump_add_to_database __bp_interactive_mode
```

## Your Environment

* Bash-it version used:
* List of enabled plugins, themes and aliases (use ``bash-it show (plugins/themes/aliases)``):
* ``bash-it doctor`` output:
* Bash version: 5.0.17
* Operating System and version: ubuntu20.94

## Your Bash Config File

```bash
  # Your bash config file should be here
```

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
git-vendor-name: preexec
git-vendor-dir: vendor/github.com/rcaloras/bash-preexec
git-vendor-repository: https://github.com/rcaloras/bash-preexec
git-vendor-ref: fd2ffa8876d3940c97ffdc3cc807e43277cf72da

## Motivation and Context
A few upstream changes, notably proper handling of previously-undefined `$PROMPT_COMMAND` and properly handling `extdebug` to prevent subshells from inheriting the `DEBUG` trap.

Alsö, use `_bash_it_library_finalize_hook` to avoid touching `$PROMPT_COMMAND` until the last possible moment, which fixes #2070. 

## How Has This Been Tested?
Live on my main branch for ages.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: lib/preexec.bash [hunk 1]
```diff
@@ -20,10 +19,9 @@ function __bp_require_not_readonly() { :; }
 
 # Disable trap DEBUG on subshells - https://github.com/Bash-it/bash-it/pull/1040
 __bp_enable_subshells= # blank
-set +T
```
Comment (NoahGorny) [inline lib/preexec.bash:23]: why did you remove this?

Comment (gaelicWizard) [reply lib/preexec.bash:23]: It's handled upstream now, which I believe I put in the commit message.

It alsö breaks BATS so needs to be handled properly, not just forced, and upstream uses the variable to control it.

Comment (NoahGorny) [reply lib/preexec.bash:23]: I see, in any case, we should note that in the commit message, and probably update the comment in line 21 (disable trap via upstream preexec)

Comment (gaelicWizard) [reply lib/preexec.bash:23]: Yeah, I squashed that commit...I had a note but then lost it 😃 

I'll update the note and the comment 👍

Comment (gaelicWizard) [reply lib/preexec.bash:23]:
```suggestion
# For performance, testing, and to avoid unexpected behavior: disable DEBUG traps in subshells.
# See bash-it/bash-it#1040 and rcaloras/bash-preexec#26
: "${__bp_enable_subshells:=}" # blank
```

Comment (gaelicWizard) [reply lib/preexec.bash:23]: Done and pushed

**Review by NoahGorny — APPROVED** [review]
LGTM
